src/sentry/auth/authenticators/u2f.py

```diff
@@ -88,36 +100,65 @@ def get_u2f_devices(self):
             # XXX: The previous version of python-u2flib-server didn't store
             # the `version` in the device binding. Defaulting to `U2F_V2` here
             # so that we don't break existing u2f registrations.
-            data["binding"].setdefault("version", "U2F_V2")
-            rv.append(DeviceRegistration(data["binding"]))
+            if type(data["binding"]) == AuthenticatorData:
+                rv.append(data["binding"])
+            else:
+                data["binding"].setdefault("version", "U2F_V2")
+                rv.append(DeviceRegistration(data["binding"]))
         return rv
 
     def remove_u2f_device(self, key):
         """Removes a U2F device but never removes the last one.  This returns
         False if the last device would be removed.
         """
-        devices = [x for x in self.config.get("devices") or () if x["binding"]["keyHandle"] != key]
+        devices = [
+            x
+            for x in self.config.get("devices") or ()
+            if (
+                urlsafe_b64encode(x["binding"].credential_data.credential_id).decode("ascii") != key
+                if type(x["binding"]) == AuthenticatorData
+                else x["binding"]["keyHandle"] != key
+            )
+        ]
         if devices:
             self.config["devices"] = devices
             return True
         return False
 
     def get_device_name(self, key):
         for device in self.config.get("devices") or ():
-            if device["binding"]["keyHandle"] == key:
+            if not type(device["binding"]) == AuthenticatorData:
+                if device["binding"]["keyHandle"] == key:
+                    return device["name"]
+            elif (
+                urlsafe_b64encode(device["binding"].credential_data.credential_id).decode("ascii")
```

Comment:
it seems this same line is copied from above, is there an opportunity to add it onto the class so you don't have to decode it everywhere?